src/Domain/Tool/Phpunit/PhpUnitTask.php

```diff
@@ -132,21 +132,49 @@ private function setTestSuite(): void {
    * ```
    */
   private function setCoverageFilter(): void {
-    $directory = $this->doc
-      ->createElement('directory', $this->getPath());
-    $exclude = $this->doc
-      ->createElement('exclude');
+
+    // Removing default "whitelist" element.
+    $whitelist = $this->xpath
+      ->query('//phpunit/filter/whitelist')
+      ->item(0);
+    assert ($whitelist instanceof \DOMElement);
+    $whitelist->parentNode->removeChild($whitelist);
+
+    // Creating new "whitelist" element.
+    $whitelist = $this->doc->createElement('whitelist');
+
+    // Excluding tests directories.
+    $exclude = $this->doc->createElement('exclude');
+
     $exclude_directory = $this->doc
-      ->createElement('directory', "{$this->getPath()}/tests");
+      ->createElement('directory', '../modules/*/tests');
     $exclude->appendChild($exclude_directory);
+
+    $exclude_directory = $this->doc
+      ->createElement('directory', '../modules/*/*/tests');
+    $exclude->appendChild($exclude_directory);
+
+    $exclude_directory = $this->doc
+      ->createElement('directory', '../*/contrib/*/tests');
+    $exclude->appendChild($exclude_directory);
+
+    // Appending the excluded directories to "whitelist" element.
+    $whitelist->appendChild($exclude);
+
+    // Adding suffixes to "whitelist" element.
+    $suffixes = ['.php', '.inc', '.module', '.install'];
```

Comment:
Since this can be used for themes or profiles, we should add:

* `.theme`
* `.profile`
* `.engine`

```suggestion
    // Adding suffixes to "whitelist" element.
    $suffixes = ['.php', '.inc', '.module', '.install', '.theme', '.profile', 'engine'];
```

Here's the list from phpstan-drupal, which I think the additions would cover

```
	fileExtensions:
		- module
		- theme
		- inc
		- install
		- profile
		- engine
```